indexing-service/src/main/java/io/druid/indexing/common/task/RealtimeIndexTask.java

```diff
@@ -325,6 +325,10 @@ public String getVersion(final Interval interval)
       }
     }
 
+    if (fireDepartment.getMetrics().failedHandoffs() > 0 ) {
```

Comment:
This won't give the right behavior either, since one of the intermediate merge-and-pushes could fail, get retried, and then work. In that case the task should still exit successfully, but with this patch it'll exit failurely. Ideally what should happen is that any handoffs that fail while the plumber is "shuttingDown" should cause finishJob to throw an exception.

I also think using metrics for this is kind of strange since IMO metrics should be write-only.